src/targets/gpu/prefuse_ops.cpp

```diff
@@ -181,17 +182,20 @@ struct find_gemm_softmax_gemm
         auto ins       = r.result;
         auto gemm2_ins = r.instructions["gemm2"];
         auto gemm1_ins = r.instructions["gemm1"];
-        auto scale_lit = r.instructions["scale"];
 
         float scale = 1.0;
-        scale_lit->eval().visit([&](const auto s) {
-            // CK only supports single-valued scale
-            if(std::all_of(
-                   s.begin() + 1, s.end(), [&](auto v) { return float_equal(v, s.front()); }))
-                scale = s.front();
-            else
-                return;
-        });
+        if(contains(r.instructions, "scale"))
+        {
+            auto scale_lit = r.instructions["scale"];
+            scale_lit->eval().visit([&](const auto s) {
+                // CK only supports single-valued scale
+                if(std::all_of(
```

Comment:
It might be better to flip the logic so that it returns if scale values are different.

Reply:
added

Reply:
I was thinking something like this
```suggestion
                if(not std::all_of(....) return;
                scale = s.front();
```

Reply:
But thats inside a visit, that return will not exit the apply method for this matcher

Reply:
oh yeah I saw that too. But looks like you rewrote it and looks fine now.